elements/rh-switch/rh-switch.ts

```diff
@@ -0,0 +1,194 @@
+
+import { InternalsController } from '@patternfly/pfe-core/controllers/internals-controller.js';
+import { SlotController } from '@patternfly/pfe-core/controllers/slot-controller.js';
+
+import { observed } from '@patternfly/pfe-core/decorators/observed.js';
+
+import { DirController } from '../../lib/DirController.js';
+import { colorContextConsumer, type ColorTheme } from '../../lib/context/color/consumer.js';
+
+
+import styles from './rh-switch.css';
+
+/**
+ * A switch toggles the state of a setting (between on and off). Switches and checkboxes can often be used interchangeably, but the switch provides a more explicit, visible representation on a setting.
+ *
+ * @summary  A switch toggles the state of a setting (between on and off).
+ *
+ * @cssprop --rh-switch-unchecked - The background color of the switch when it is unchecked.
+ * @cssprop --rh-switch-checked - The background color of the switch when it is checked.
+ * @cssprop --rh-switch-disabled - The background color of the switch when it is disabled.
+ */
+@customElement('rh-switch')
+export class RhSwitch extends LitElement {
+  static readonly styles = [styles];
+
+  static readonly formAssociated = true;
+
+  @property({ reflect: true }) label?: string;
+
+  @property({ reflect: true, type: Boolean, attribute: 'show-check-icon' }) showCheckIcon = false;
+
+  @observed
+  @property({ reflect: true, type: Boolean }) checked = false;
+
+  @property({ reflect: true, type: Boolean }) disabled = false;
+
+  @property({ reflect: true, attribute: 'accessible-label' }) accessibleLabel?: string;
+
+  @property({ reflect: true, type: Boolean }) reversed = false;
+
+  /**
+   * Sets color theme based on parent context
+   */
+  @colorContextConsumer() private on?: ColorTheme;
+
+  #internals = InternalsController.of(this, { role: 'switch' });
+
+  #slots = new SlotController(this);
+
+  #dir = new DirController(this);
+
+  #internalColorPalette?: string | null;
+
+  get labels(): NodeListOf<HTMLLabelElement> {
+    return this.#internals.labels as NodeListOf<HTMLLabelElement>;
+  }
+
+  formDisabledCallback(disabled: boolean) {
+    this.disabled = disabled;
+  }
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    this.tabIndex = 0;
+    this.addEventListener('click', this.#onClick);
+    this.addEventListener('keyup', this.#onKeyup);
+    this.addEventListener('keydown', this.#onKeyDown);
+    this.#updateLabels();
+  }
+
+  willUpdate() {
+    /*
+     * TEMPORARY: this fixes the need to access the parents color-palette in order to get the `lightest`
+     * value.  This fix will only update the component when switching between light and dark themes as
+     * thats when the consumer requests an update.  Switching between lighter -> light for example will
+     * not trigger the component to update at this time.
+     */
+    this.#internalColorPalette = this.closest('[color-palette]')?.getAttribute('color-palette');
```

Comment:
there is a need for specific color-palette because to make given a specific color way accessible https://github.com/RedHat-UX/red-hat-design-system/pull/1513#discussion_r1601747740

```
:host(:disabled) .color-palette-dark {
  --_switch-track-background-color: var(--rh-color-gray-60, #4d4d4d);
}
```

which uses the `.color-palette-dark`.

Reply:
related to: https://github.com/RedHat-UX/red-hat-design-system/pull/1395

Reply:
So why not make it that color for all dark contexts?

Reply:
@marionnegp thoughts on this?

Reply:
Alternatively, solve #1395 without too-too many breaking changes, and block this pr on it

Reply:
@marionnegp Hard coded the disabled values 7e7f15a220c90bb78c1ed1a542b5d20500a42ee0 as we were discussing in office hours.

Please take a look at the disabled switches on the color context demo and let me know if this passes design muster.

https://deploy-preview-1513--red-hat-design-system.netlify.app/elements/switch/demo/color-context/

Reply:
resolved in @marionnegp commit 5d97936